### PR TITLE
Add Engstudio Block67 HS

### PR DESCRIPTION
**Pull request by mrnoisytiger**
## Description

Add Engstudio Block67 HS

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fjlabs/block67hs/config.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x7074 // FJLabs
+#define PRODUCT_ID   0x1021 // Block67 Hotswap
+#define DEVICE_VER   0x0001 // Version 1
+#define MANUFACTURER    FJLabs
```
Comment (zvecr) [inline keyboards/fjlabs/block67hs/config.h:26]: You might want to re-read the PR checklist, as this doesn't even compile against master (and wont have done for a fair while now).

Comment (mrnoisytiger) [reply keyboards/fjlabs/block67hs/config.h:26]: **sigh**, pushing 2 year old code up always begets some problems. I'm still running a much older QMK locally, so I suppose that's why it compiled perfectly here. 

Will update for this new `info.json`...

File: keyboards/fjlabs/block67hs/block67hs.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, K2E, \
+    K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E, \
+    K40, K41, K42,                K46,                K4A, K4B, K4C, K4D, K4E \
+) { \
+    { K00, K01, K02, K03,   K04,     K05,   K06,   K07,   K08,   K09,   K0A, K0B, K0C, K0D, K0E }, \
+    { K10, K11, K12, K13,   K14,     K15,   K16,   K17,   K18,   K19,   K1A, K1B, K1C, K1D, K1E }, \
+    { K20, K21, K22, K23,   K24,     K25,   K26,   K27,   K28,   K29,   K2A, K2B, K2C, K2D, K2E }, \
+    { K30, K31, K32, K33,   K34,     K35,   K36,   K37,   K38,   K39,   K3A, K3B, K3C, K3D, K3E }, \
+    { K40, K41, K42, KC_NO, KC_NO,   KC_NO, K46,   KC_NO, KC_NO, KC_NO, K4A, K4B, K4C, K4D, K4E }  \
+}
```
Comment (waffle87) [inline keyboards/fjlabs/block67hs/block67hs.h:32]: This belongs in `info.json` now. This file can be removed after migrating.

File: keyboards/fjlabs/block67hs/block67hs.c [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (waffle87) [inline keyboards/fjlabs/block67hs/block67hs.c:13]: This file can also be removed.

File: keyboards/fjlabs/block67hs/config.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *         ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+*/
+
```
Comment (waffle87) [inline keyboards/fjlabs/block67hs/config.h:34]:
```suggestion
```

File: keyboards/fjlabs/block67hs/config.h [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+#define MATRIX_ROW_PINS { F4, F1, F7, F6, F5 }
+#define MATRIX_COL_PINS { F0, B1, B2, B3, B7, D0, D1, D2, D3, D5, D7, B4, B5, B6, C6 }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
```
Comment (waffle87) [inline keyboards/fjlabs/block67hs/config.h:39]: Same here — this needs to be in `info.json`. And this file can then be removed, unless you have need for the `LOCKING` defines..

File: keyboards/fjlabs/block67hs/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency
+F_CPU = 8000000
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/fjlabs/block67hs/rules.mk:9]:
```suggestion
# Processor frequency
F_CPU = 8000000
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**